From 1cede8cc5d5b5ae2bc88a34c88453ee839d11ff0 Mon Sep 17 00:00:00 2001
From: Hai Zhang <zhanghai@google.com>
Date: Wed, 16 Jun 2021 00:20:52 +0000
Subject: [PATCH] DO NOT MERGE Add cross-user check for getDefaultSmsPackage().

Bug: 177927831
Test: atest RoleSecurityTest
Change-Id: I1254804fb72a299e782d45f938acdf979a82f904
(cherry picked from commit cf1bd25a37123624688f9965608a48d362ab4eb0)
---
 .../java/com/android/server/role/RoleManagerService.java    | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/services/core/java/com/android/server/role/RoleManagerService.java b/services/core/java/com/android/server/role/RoleManagerService.java
index 392792dbae69..b75bce833e04 100644
--- a/services/core/java/com/android/server/role/RoleManagerService.java
+++ b/services/core/java/com/android/server/role/RoleManagerService.java
@@ -662,6 +662,12 @@ public class RoleManagerService extends SystemService implements RoleUserState.C
 
         @Override
         public String getDefaultSmsPackage(int userId) {
+            userId = handleIncomingUser(userId, false, "getDefaultSmsPackage");
+            if (!mUserManagerInternal.exists(userId)) {
+                Slog.e(LOG_TAG, "user " + userId + " does not exist");
+                return null;
+            }
+
             long identity = Binder.clearCallingIdentity();
             try {
                 return CollectionUtils.firstOrNull(
-- 
2.17.1

